Tasks/DeployVisualStudioTestAgent/TestAgentConfiguration.ps1

```diff
@@ -1,4 +1,143 @@
+		return $true
+	}
+	elseif($autoadminLogon -eq "1")
+	{
+		return $false
+	}
+
+	Write-Verbose -Message("Registry path {0} found, but AutoAdminLogon key is not enabled." -f $registryPath) -Verbose
+	return $true
+}
+
+function isShowLogonMessagePolicyEnabled([int] $OSType)
+{
+	#check 64 bit or 32 bit
+	if ($OSType -eq 64)
+	{
+		$registryPath = "HKLM:\SOFTWARE\Wow6432Node\Microsoft\Windows\CurrentVersion\policies\system"
+	}
+	else
+	{
+		$registryPath = "HKLM:\SOFTWARE\Microsoft\Windows\CurrentVersion\policies\system"
+	}
+
+	if (-not (Test-Path $registryPath))
+	{
+		Write-Verbose -Message("Registry path {0} not found" -f $registryPath) -Verbose
+		return $false
+	}
+	
+	$legalCaption = (Get-ItemProperty $registryPath -ErrorAction SilentlyContinue).legalnoticecaption
+	if (($legalCaption) -and ($legalCaption.Length -gt 0))
+	{
+		Write-Verbose -Message("Registry path {0} found, but legalnoticecaption key is set to some value." -f $registryPath) -Verbose
+		return $true
+	}
+
+	$legalText = (Get-ItemProperty $registryPath -ErrorAction SilentlyContinue).legalnoticetext
+	if (($legalText) -and ($legalText.Length -gt 0))
+	{
+		Write-Verbose -Message("Registry path {0} found, but legalnoticetext key is set to some value." -f $registryPath) -Verbose
+		return $true
+	}
+
+	return $false
+}
+
+function isShowLogonMessageEnabled([int] $OSType)
+{
+	#check 64 bit or 32 bit
+	if ($OSType -eq 64)
+	{
+		$registryPath = "HKLM:\SOFTWARE\Wow6432Node\Microsoft\Windows NT\CurrentVersion\Winlogon"
+	}
+	else
+	{
+		$registryPath = "HKLM:\SOFTWARE\Microsoft\Windows NT\CurrentVersion\Winlogon"
+	}
+
+	if (-not (Test-Path $registryPath))
+	{
+		Write-Verbose -Message("Registry path {0} not found" -f $registryPath) -Verbose
+		return $false
+	}
+	
+	$legalCaption = (Get-ItemProperty $registryPath -ErrorAction SilentlyContinue).LegalNoticeCaption
+	if (($legalCaption -ne $null) -and ($legalCaption.Length -gt 0))
+	{
+		Write-Verbose -Message("Registry path {0} found, but LegalNoticeCaption is set to some value." -f $registryPath) -Verbose
+		return $true
+	}
+
+	$legalText = (Get-ItemProperty $registryPath -ErrorAction SilentlyContinue).LegalNoticeText
+	if (($legalText -ne $null) -and ($legalText.Length -gt 0))
+	{
+		Write-Verbose -Message("Registry path {0} found, but LegalNoticeText is set to some value." -f $registryPath) -Verbose
+		return $true
+	}
+
+	return $false
+}
+
+function Update-RebootCount([string] $environmentURL)
+{
+	if ([IntPtr]::Size -eq 8)
+	{
+		$testAgentRegPath = "HKLM:\SOFTWARE\Wow6432Node\Microsoft\TestAgentConfig"
+	}
+	else
+	{
+		$testAgentRegPath = "HKLM:\SOFTWARE\Microsoft\TestAgentConfig" 
+	}
+
+	if (-not (Test-Path $testAgentRegPath))
+	{
+		New-Item -Path $testAgentRegPath -Force | Out-Null
+		Write-Verbose -Message ("Updating machine reboot count to 1") -Verbose
+		New-ItemProperty -Path $testAgentRegPath -Name "MachineRebootCount" -Value 1 -PropertyType DWord -Force | Out-Null
```

Comment:
we are resetting reboot count every time there is new environment URL, and same key and name is being reused so no need to delete

---
In reply to: [103856036](https://github.com/Microsoft/vsts-tasks/pull/3701#discussion_r103856036) [](ancestors = 103856036)